src/components/CartOrderForm.js

```diff
@@ -111,14 +116,36 @@ export const CartOrderForm = ({
             inputProps={user.userType === 'user' ? cartFormClasses.dateProps : {}}
           />
 
-          <FormControl align='left' fullWidth={true} error={deliveryMethodError} sx={{ my: 1 }}>
-            <FormLabel id='demo-row-radio-buttons-group-label'>Método de entrega</FormLabel>
-            <RadioGroup row aria-labelledby='demo-row-radio-buttons-group-label' name='row-radio-buttons-group' onChange={handleRadioClick}>
-              <FormControlLabel value='delivery' control={<Radio />} label='Entrega' checked={deliveryMethod === 'delivery'} />
-              <FormControlLabel value='takeAway' control={<Radio />} label='Take Away' checked={deliveryMethod === 'takeAway'} />
-            </RadioGroup>
-          </FormControl>
+          <Box sx={{ display: { xs: 'block', md: 'flex' }, mb: 2 }}>
+            <FormControl align='left' fullWidth={true} error={deliveryMethodError} sx={{ my: 1 }}>
+              <FormLabel id='demo-row-radio-buttons-group-label'>Método de entrega</FormLabel>
+              <RadioGroup row aria-labelledby='demo-row-radio-buttons-group-label' name='row-radio-buttons-group' onChange={handleRadioClick}>
+                <FormControlLabel value='delivery' control={<Radio />} label='Entrega' checked={deliveryMethod === 'delivery'} />
+                <FormControlLabel value='takeAway' control={<Radio />} label='Take Away' checked={deliveryMethod === 'takeAway'} />
+              </RadioGroup>
+            </FormControl>
 
+            {deliveryMethod === 'delivery' && user.userType === 'admin' && (
+              <FormControlLabel
+                control={<Switch checked={haveExtraFee} onChange={() => setHaveExtraFee(!haveExtraFee)} inputProps={{ 'aria-label': 'controlled' }} />}
+                label='Definir taxa de entrega'
+                sx={{ width: '100%', pt: { xs: 0, md: 3 } }}
+              />
+            )}
+          </Box>
+          {user.userType === 'admin' && haveExtraFee && (
+            <TextField
+              label='Taxa de entrega'
+              type='text'
+              variant='outlined'
+              fullWidth
+              required
+              sx={cartFormClasses.formField}
+              onChange={(e) => validateDeliveryFee(e)}
+              error={deliveryPriceError}
+              value={deliveryPrice}
+            />
+          )}
```

Comment:
having admin and final user code on the same place isn't ideal because it can increase final bundle size which may affect performance, load time, etc.

I don't think you have to worry right now buuuut

Reply:
So in that case the best would be to have it in separated files and then just import it there? Can you explain how would you solve it?

Reply:
I would solve having 2 different applications or at least 2 different routes. Componentizing stuff and lazy loading would also solve

Reply:
Yes i understand. I implemented lazy loading and will componentize more stuff. Merci.